### PR TITLE
Remove visitNewArray to deprecate getVisitorState Usage

### PR DESCRIPTION
**Pull request by zcai1**
This is to resolve the latest compilation issue in https://github.com/opprop/checker-framework-inference/pull/391

### CODE REVIEW

File: src/main/java/universe/UniverseAnnotatedTypeFactory.java [hunk 1]
```diff
@@ -155,92 +149,6 @@ public UniversePropagationTreeAnnotator(AnnotatedTypeFactory atypeFactory) {
...
-                    // Somebody else will complain.
-                    post = prev;
-                }
-            } else {
-                // No context is available - simply use what we have.
-                post = prev;
-            }
-
-            // Below line is the only difference from super implementation
-            applyImmutableIfImplicitlyBottom(componentType);
```
Comment (wmdietl) [inline src/main/java/universe/UniverseAnnotatedTypeFactory.java:236]: Can you add a test case that uses an "implicitly bottom" (whatever that was...) component type in an array to see whether this code actually did anything?

Comment (wmdietl) [reply src/main/java/universe/UniverseAnnotatedTypeFactory.java:236]: @zcai1 I think you're looking into how to properly implement this. Should we merge this PR and you file an issue/open a new PR with test case and fixes?

Comment (wmdietl) [reply src/main/java/universe/UniverseAnnotatedTypeFactory.java:236]: @zcai1 I think you're looking into how to properly implement this. Should we merge this PR and you file an issue/open a new PR with test case and fixes?

Comment (zcai1) [reply src/main/java/universe/UniverseAnnotatedTypeFactory.java:236]: @wmdietl yes, I think we should fix this in a separate PR.

**Pull request merged**